### DISCUSSION TITLE
Which algorithm does the JavaScript Array.sort() or Array.sort((a,b)=>a-b) function use?

### DISCUSSION THREAD

**Discussion by harshal255** [community/community#65230] [Discussions]
### Select Topic Area

Question

### Which algorithm does the JavaScript Array.sort() or Array.sort((a,b)=>a-b) function use?
```
//Array sort method -> change original array

const numbersArr = [3, 10, 4, 21, 5, 9, 2, 6, 5, 3, 5];

// Ascending Order
numbersArr.sort((a, b) => a - b);
console.log(numbersArr); 

// Descending Order
numbersArr.sort((a, b) => b - a);
console.log(numbersArr); 

//Alphabetical Order
numbersArr.sort();
console.log(numbersArr);
```

**Comment by queenofcorgis** [comment]
The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was closed because it contained content that violates our Community Code of Conduct, including posting the same comment in multiple places. As a result, we have hidden the duplicate comments, and we ask that you no longer cross-post.

For additional guidance, please review our [Community on Discussions Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service).

**Comment by mohamed653** [comment]
The exact algorithm used can vary depending on the JavaScript engine. 
for some browsers like Chrome engine **V8**  uses a combination of **quicksort** and **insertion sort**
